app/controllers/ApplicationController.scala

```diff
@@ -60,8 +59,8 @@ trait ApplicationController extends BaseController {
   }
 
   def applicationStatus(applicationId: String) = Action.async { implicit request =>
-    appRepository.findProgress(applicationId).map { result =>
-      Ok(Json.toJson(ApplicationStatusOrder.getStatus(result)))
+    appRepository.findStatus(applicationId).map { result =>
```

Comment:
I would call ApplicationController.applicationStatus() the same as ApplicationRepository.findStatus. My suggestion would be aplicationStatusDetails as this is what we are dealing with, and status or applicationStatus refers more to the status string.